src/viam/services/mlmodel/utils.py

```diff
@@ -37,7 +39,14 @@ def make_ndarray(flat_data, dtype, shape):
         """Takes flat data (protobuf RepeatedScalarFieldContainer | bytes) to output an ndarray
         of appropriate dtype and shape"""
         make_array = np.frombuffer if dtype == np.int8 or dtype == np.uint8 else np.array
-        return make_array(flat_data, dtype).reshape(shape)
+        # as per proto, int16 and uint16 are stored as uint32. As of numpy v2, this creates
+        # some strange interactions with negative values for int16 specifically. creating
+        # our array as a uint32 array initially and then casting to int16 solves this.
```

Comment:
Mild preference to clarify what strange interactions you saw. but I guess it's not so important if they're now fixed. Leave as-is if it's too hard to describe them.

Is it worth making a unit test for this? "No" is a fine answer, but I figured I'd ask.

Reply:
We already have unit tests for all types of np array conversions, including i16 arrays with negative values. I'm open to the idea of adding new ones but I think all our cases are covered, unless I'm missing something!

Reply:
clarification on behavior added!

Reply:
...if all the cases are covered, did you find this problem because one of the unit tests failed? I might have misunderstood.

Reply:
Ahh no, the problem is we don't currently support/test python 3.13, and continue to use numpy1.26 in 3.8 through 3.12. 3.13 requires numpy 2.0 or higher, which is where the issue arises.

If we were to have added 3.13 support without making this change, at that point tests would start failing.

Reply:
That makes sense! I'm sure you'll add tests for 3.13 eventually, but it might be too early for that now.